### PR TITLE
[NAE-1962] Event properties

### PR DESCRIPTION
**Pull request by Retoocs**
# Description

This changes add parameters into action events

Implements [NAE-1962]

## Dependencies

No new dependencies were introduced

### Third party dependencies

No new dependencies were introduced

### Blocking Pull requests

There are no dependencies on other PR

## How Has Been This Tested?

Manually by testing process and by unit tests

### Test Configuration

| Name                | Tested on |
|---------------------| --------- |
| OS                  |  Linux Mint 21.1 Cinnamon |
| Runtime             |  Java 11  |
| Dependency Manager  |   Maven 3.8.1  |
| Framework version   |    Spring Boot 2.7.8  |
| Run parameters      |           |
| Other configuration |           |

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My changes have been checked, personally or remotely, with @...
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have resolved all conflicts with the target branch of the PR
- [x] I have updated and synced my code with the target branch
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing tests pass locally with my changes:
    - [x] Lint test
    - [x] Unit tests
    - [x] Integration tests
- [ ] I have checked my contribution with code analysis tools:
    - [ ] [SonarCloud](https://sonarcloud.io/project/overview?id=netgrif_application-engine)
    - [ ] [Snyk](https://app.snyk.io/org/netgrif/project/ea82b3b8-658d-41f5-89a7-76cd600da01f)
- [x] I have made corresponding changes to the documentation:
    - [x] Developer documentation
    - [x] User Guides
    - [x] Migration Guides

[NAE-1962]: https://netgrif.atlassian.net/browse/NAE-1962?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/main/java/com/netgrif/application/engine/petrinet/service/PetriNetService.java [hunk 1]
```diff
@@ -558,7 +568,7 @@ public void runActions(List<Action> actions, PetriNet petriNet) {
         log.info("Running actions of net [" + petriNet.getStringId() + "]");
 
         actions.forEach(action -> {
-            actionsRunner.run(action, null, petriNet.getFunctions());
+            actionsRunner.run(action, null, null, petriNet.getFunctions());
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/petrinet/service/PetriNetService.java:571]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/DataService.java [hunk 2]
```diff
@@ -419,8 +438,13 @@ public FileFieldInputStream getFileByTask(String taskId, String fieldId, boolean
 
     @Override
     public FileFieldInputStream getFileByTaskAndName(String taskId, String fieldId, String name) {
+        return getFileByTaskAndName(taskId, fieldId, name, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/DataService.java:444]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/DataService.java [hunk 3]
```diff
@@ -463,7 +497,12 @@ public FileFieldInputStream getFileByName(Case useCase, FileListField field, Str
 
     @Override
     public FileFieldInputStream getFile(Case useCase, Task task, FileField field, boolean forPreview) {
-        runGetActionsFromFileField(field.getEvents(), useCase);
+        return getFile(useCase, task, field, forPreview, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/DataService.java:503]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/DataService.java [hunk 4]
```diff
@@ -563,6 +602,11 @@ public InputStream download(String url) throws IOException {
 
     @Override
     public SetDataEventOutcome saveFile(String taskId, String fieldId, MultipartFile multipartFile) {
+        return saveFile(taskId, fieldId, multipartFile, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/DataService.java:608]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/DataService.java [hunk 5]
```diff
@@ -573,11 +617,16 @@ public SetDataEventOutcome saveFile(String taskId, String fieldId, MultipartFile
         } else {
             saveLocalFile(useCase, field, multipartFile);
         }
-        return new SetDataEventOutcome(useCase, task, getChangedFieldByFileFieldContainer(fieldId, task, useCase));
+        return new SetDataEventOutcome(useCase, task, getChangedFieldByFileFieldContainer(fieldId, task, useCase, params));
     }
 
     @Override
     public SetDataEventOutcome saveFiles(String taskId, String fieldId, MultipartFile[] multipartFiles) {
+        return saveFiles(taskId, fieldId, multipartFiles, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/DataService.java:628]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/DataService.java [hunk 6]
```diff
@@ -675,6 +724,11 @@ protected boolean deleteRemote(Case useCase, FileListField field, String name) {
 
     @Override
     public SetDataEventOutcome deleteFile(String taskId, String fieldId) {
+        return deleteFile(taskId, fieldId, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/DataService.java:730]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/DataService.java [hunk 7]
```diff
@@ -703,6 +757,11 @@ private ImmutablePair<Case, FileField> getCaseAndFileField(String taskId, String
 
     @Override
     public SetDataEventOutcome deleteFileByName(String taskId, String fieldId, String name) {
+        return deleteFileByName(taskId, fieldId, name, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/DataService.java:763]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/EventService.java [hunk 8]
```diff
@@ -101,17 +101,22 @@ public List<EventOutcome> processDataEvents(Field field, DataEventType actionTri
             return Collections.emptyList();
         }
 
-        return runEventActions(useCase, task, fieldActions, actionTrigger);
+        return runEventActions(useCase, task, fieldActions, actionTrigger, params);
     }
 
     @Override
     public void runEventActionsOnChanged(Task task, SetDataEventOutcome outcome, DataEventType trigger) {
+        runEventActionsOnChanged(task, outcome, trigger, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/EventService.java:109]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java [hunk 9]
```diff
@@ -121,40 +121,60 @@ public void setElasticTaskService(IElasticTaskService elasticTaskService) {
 
     @Override
     public List<AssignTaskEventOutcome> assignTasks(List<Task> tasks, IUser user) throws TransitionNotExecutableException {
+        return assignTasks(tasks, user, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java:124]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java [hunk 10]
```diff
@@ -121,40 +121,60 @@ public void setElasticTaskService(IElasticTaskService elasticTaskService) {
...
         for (Task task : tasks) {
-            outcomes.add(assignTask(task, user));
+            outcomes.add(assignTask(task, user, params));
         }
         return outcomes;
     }
 
     @Override
     public AssignTaskEventOutcome assignTask(LoggedUser loggedUser, String taskId) throws TransitionNotExecutableException {
+        return assignTask(loggedUser, taskId, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java:138]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java [hunk 11]
```diff
@@ -121,40 +121,60 @@ public void setElasticTaskService(IElasticTaskService elasticTaskService) {
...
         }
 
         IUser user = getUserFromLoggedUser(loggedUser);
-        return assignTask(taskOptional.get(), user);
+        return assignTask(taskOptional.get(), user, params);
     }
 
     @Override
     public AssignTaskEventOutcome assignTask(String taskId) throws TransitionNotExecutableException {
+        return assignTask(taskId, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java:154]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java [hunk 12]
```diff
@@ -121,40 +121,60 @@ public void setElasticTaskService(IElasticTaskService elasticTaskService) {
...
+    @Override
+    public AssignTaskEventOutcome assignTask(String taskId, Map<String, String> params) throws TransitionNotExecutableException {
         LoggedUser user = userService.getLoggedOrSystem().transformToLoggedUser();
-        return assignTask(user, taskId);
+        return assignTask(user, taskId, params);
     }
 
     @Override
     public AssignTaskEventOutcome assignTask(Task task, IUser user) throws TransitionNotExecutableException {
+        return assignTask(task, user, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java:165]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java [hunk 13]
```diff
@@ -186,21 +206,36 @@ protected Case assignTaskToUser(IUser user, Task task, String useCaseId) throws
 
     @Override
     public List<FinishTaskEventOutcome> finishTasks(List<Task> tasks, IUser user) throws TransitionNotExecutableException {
+        return finishTasks(tasks, user, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java:209]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java [hunk 14]
```diff
@@ -186,21 +206,36 @@ protected Case assignTaskToUser(IUser user, Task task, String useCaseId) throws
...
         for (Task task : tasks) {
-            outcomes.add(finishTask(task, user));
+            outcomes.add(finishTask(task, user, params));
         }
         return outcomes;
     }
 
     @Override
     public FinishTaskEventOutcome finishTask(String taskId) throws IllegalArgumentException, TransitionNotExecutableException {
+        return finishTask(taskId, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java:223]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java [hunk 15]
```diff
@@ -186,21 +206,36 @@ protected Case assignTaskToUser(IUser user, Task task, String useCaseId) throws
...
+    @Override
+    public FinishTaskEventOutcome finishTask(String taskId, Map<String, String> params) throws IllegalArgumentException, TransitionNotExecutableException {
         LoggedUser user = userService.getLoggedOrSystem().transformToLoggedUser();
-        return finishTask(user, taskId);
+        return finishTask(user, taskId, params);
     }
 
     @Override
     public FinishTaskEventOutcome finishTask(LoggedUser loggedUser, String taskId) throws IllegalArgumentException, TransitionNotExecutableException {
+        return finishTask(loggedUser, taskId, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java:234]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java [hunk 16]
```diff
@@ -254,39 +294,54 @@ public FinishTaskEventOutcome finishTask(Task task, IUser user) throws Transitio
 
     @Override
     public List<CancelTaskEventOutcome> cancelTasks(List<Task> tasks, IUser user) {
+        return cancelTasks(tasks, user, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java:297]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java [hunk 17]
```diff
@@ -254,39 +294,54 @@ public FinishTaskEventOutcome finishTask(Task task, IUser user) throws Transitio
...
         for (Task task : tasks) {
-            outcomes.add(cancelTask(task, user));
+            outcomes.add(cancelTask(task, user, params));
         }
         return outcomes;
     }
 
     @Override
     public CancelTaskEventOutcome cancelTask(LoggedUser loggedUser, String taskId) {
+        return cancelTask(loggedUser, taskId, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java:311]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java [hunk 18]
```diff
@@ -303,16 +358,24 @@ public CancelTaskEventOutcome cancelTask(Task task, IUser user) {
      */
     @Override
     public void cancelTasksWithoutReload(Set<String> transitions, String caseId) {
+        cancelTasksWithoutReload(transitions, caseId, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java:361]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java [hunk 19]
```diff
@@ -338,6 +401,11 @@ private Task returnTokens(Task task, String useCaseId) {
 
     @Override
     public DelegateTaskEventOutcome delegateTask(LoggedUser loggedUser, String delegatedId, String taskId) throws TransitionNotExecutableException {
+        return delegateTask(loggedUser, delegatedId, taskId, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/TaskService.java:404]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/WorkflowService.java [hunk 20]
```diff
@@ -247,28 +247,52 @@ private List<String> getExistingUsers(UserListFieldValue userListValue) {
...
-            return this.createCase(netId, resolveDefaultCaseTitle(netId, locale), color, user);
+            return this.createCase(netId, resolveDefaultCaseTitle(netId, locale, params), color, user, params);
         }
-        return this.createCase(netId, title, color, user);
+        return this.createCase(netId, title, color, user, params);
+    }
+
+    @Override
+    public CreateCaseEventOutcome createCase(String netId, String title, String color, LoggedUser user, Locale locale) {
+        return this.createCase(netId, title, color, user, locale, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/WorkflowService.java:262]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/WorkflowService.java [hunk 21]
```diff
@@ -247,28 +247,52 @@ private List<String> getExistingUsers(UserListFieldValue userListValue) {
...
+        PetriNet net = petriNetService.getNewestVersionByIdentifier(identifier);
+        if (net == null) {
+            throw new IllegalArgumentException("Petri net with identifier [" + identifier + "] does not exist.");
+        }
+        return this.createCase(net.getStringId(), title != null && !title.equals("") ? title : net.getDefaultCaseName().getTranslation(locale), color, user, params);
     }
 
     @Override
     public CreateCaseEventOutcome createCaseByIdentifier(String identifier, String title, String color, LoggedUser user, Locale locale) {
+        return this.createCase(identifier, title, color, user, locale, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/WorkflowService.java:286]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/WorkflowService.java [hunk 22]
```diff
@@ -281,19 +305,23 @@ public CreateCaseEventOutcome createCaseByIdentifier(String identifier, String t
     }
 
     public CreateCaseEventOutcome createCase(String netId, Function<Case, String> makeTitle, String color, LoggedUser user) {
+        return this.createCase(netId, makeTitle, color, user, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/WorkflowService.java:308]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/WorkflowService.java [hunk 23]
```diff
@@ -343,27 +371,37 @@ public Page<Case> findAllByAuthor(String authorId, String petriNet, Pageable pag
...
+    public DeleteCaseEventOutcome deleteCase(String caseId, Map<String, String> params) {
         Case useCase = findOne(caseId);
-        return deleteCase(useCase);
+        return deleteCase(useCase, params);
     }
 
     @Override
-    public DeleteCaseEventOutcome deleteCase(Case useCase) {
+    public DeleteCaseEventOutcome deleteCase(String caseId) {
+        return deleteCase(caseId, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/WorkflowService.java:381]: new HashMap

File: src/main/java/com/netgrif/application/engine/workflow/service/WorkflowService.java [hunk 24]
```diff
@@ -343,27 +371,37 @@ public Page<Case> findAllByAuthor(String authorId, String petriNet, Pageable pag
...
-        outcome.addOutcomes(eventService.runActions(useCase.getPetriNet().getPostDeleteActions(), null, Optional.empty()));
+        outcome.addOutcomes(eventService.runActions(useCase.getPetriNet().getPostDeleteActions(), null, Optional.empty(), params));
         addMessageToOutcome(useCase.getPetriNet(), CaseEventType.DELETE, outcome);
         historyService.save(new DeleteCaseEventLog(useCase, EventPhase.POST));
         return outcome;
     }
 
+    @Override
+    public DeleteCaseEventOutcome deleteCase(Case useCase) {
+        return deleteCase(useCase, null);
```
Comment (mazarijuraj) [inline src/main/java/com/netgrif/application/engine/workflow/service/WorkflowService.java:402]: new HashMap

**Review by renczesstefan — APPROVED** [review]

**Review by Kovy95 — APPROVED** [review]

**Review by tuplle — APPROVED** [review]

**Review by mazarijuraj — APPROVED** [review]

**Pull request merged**